lp-core-platform/lp-cp-xwiki-implementation/src/main/java/eu/learnpad/core/impl/mv/XwikiController.java

```diff
@@ -121,15 +125,42 @@ public void notifyVerification(String verificationProcessId)
         String type = this.typesMap.get(modelSetId);
         
 		boolean resultsOk = res.getFinalResult().equals(FinalResultType.OK);
+		boolean importedInTheSimulator = true;
 		if(resultsOk){
 		    if(XWikiRestUtils.isPage(RestResource.CORE_REPOSITORY_WIKI,
 	                RestResource.CORE_REPOSITORY_SPACE, modelSetId) == true){
     		    this.cw.modelSetImported(modelSetId, type);
     		    this.or.modelSetImported(modelSetId, type);
+    		    
+    		    String bpmnFileURL = XWikiRestUtils.exposeBPMNFromCoreRepository(modelSetId, type);
+    		    Collection<String> savedProcessesList = this.sim.addProcessDefinition(bpmnFileURL);
+    		    importedInTheSimulator = this.verifyImportedProcesses(savedProcessesList, bpmnFileURL);
 		    }
 		}
 		
-		this.cw.modelVerified(modelSetId, (resultsOk)?"OK":"KO");
+		// the value for message may change accordingly to eu.learnpad.cw.rest.ModelVerified
+		String message = "KO";
+		if ( resultsOk ){
```

Comment:
I don't really understand the purpose of this block.  The notification `cw.modelVerified` exists only to notify the CW that the model verification finish it's job and that everything is OK.  So no relation with SIM.

By the way, this notification `cw.modelVerified` (which should be called `cw.modelsetVerified` is not really needed anymore since we now specified that until the modelset has been verified, the Core Platform doesn't notify any component of this new modelset.  So CW will do nothing from this notification (it will reacts on notification `cw.modelsetImported` sent above).